ui_testing/testcases/basic_tests/test05_testplans.py

```diff
@@ -51,8 +51,417 @@ def test001_test_plan_delete_testunit(self):
+
+        if completed_testplans is not None:
+            self.base_selenium.LOGGER.info('Getting the first testplan')
+            completed_testplan = completed_testplans[0]
+            old_completed_testplan_name = completed_testplan['testPlanName']
+            old_completed_testplan_version = completed_testplan['version']
+            self.base_selenium.LOGGER.info(
+                'Navigating to edit page of testplan: {} with version: {}'.format(old_completed_testplan_name,
+                                                                                  old_completed_testplan_version))
+            self.test_plan.get_test_plan_edit_page(name=old_completed_testplan_name)
+
+            # go to step 2 and remove all the testunits
+            self.base_selenium.LOGGER.info('Going to step 2 to remove all the testunits from it')
+            self.test_plan.navigate_to_testunits_selection_page()
+            self.test_plan.delete_all_testunits()
+            self.test_plan.save_and_confirm_popup()
+
+            # go back to the active table
+            self.test_plan.get_test_plans_page()
+
+            # get the testplan to check its version
+            self.base_selenium.LOGGER.info('Getting the currently changed testplan to check its status and version')
+            inprogress_testplan_version, testplan_row_data_status = self.test_plan.get_testplan_version_and_status(
+                search_text=old_completed_testplan_name)
+
+            self.assertEqual(old_completed_testplan_version + 1, int(inprogress_testplan_version))
+            self.assertEqual(testplan_row_data_status, 'In Progress')
+
+    def test010_create_testplans_same_name_article_materialtype(self):
+        '''
+        LIMS-3499
+        Testing the creation of two testplans with the same name, material type
+        and article, this shouldn't happen
+        '''
+
+        # get all articles and choose a random one to take its information
+        articles = self.get_all_articles()
+        random_index = self.generate_random_number(lower=0, upper=len(articles))
+        article_data = articles[random_index]
+        self.base_selenium.LOGGER.info(
+            'A random article is chosen, its name: {} and its material type: {}'.format(article_data['name'],
+                                                                                        article_data['materialType']))
+
+        testplan_name = self.test_plan.create_new_test_plan(material_type=article_data['materialType'],
+                                                            article=article_data['name'])
+        self.base_selenium.LOGGER.info(
+            'New testplan is created successfully with name: {}, article name: {} and material type: {}'.format(
+                testplan_name, article_data['name'], article_data['materialType']))
+
+        self.base_selenium.LOGGER.info(
+            'Attempting to create another testplan with the same data as the previously created one')
```

Comment:
Just call the API and get any existing test plan then use its data to create the new one